File: app/src/main/java/lt/gecius/test_dagger2/DemoApplication.java
package lt.gecius.test_dagger2;

import android.app.Application;

public class DemoApplication extends Application {
	private ApplicationComponent component;

	@Override
	public void onCreate() {
		super.onCreate();
		component = DaggerApplicationComponent.builder()
				.applicationModule(new ApplicationModule(this))
				.build();
	}

	public ApplicationComponent getComponent() {
		return component;
	}
}

File: app/src/main/java/lt/gecius/test_dagger2/app/DemoApplication.kt
package lt.gecius.test_dagger2.app

import android.app.Application

import lt.gecius.test_dagger2.di.ApplicationComponent
import lt.gecius.test_dagger2.di.ApplicationModule
import lt.gecius.test_dagger2.di.DaggerApplicationComponent


class DemoApplication : Application() {

    lateinit var component: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()
        component = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
    }
}
